lib/game_menu.rb

```diff
@@ -0,0 +1,80 @@
+# frozen_string_literal: true
+
+require_relative '../dependencies'
+
+class GameMenu
+  class << self
+    include Validation
+    include Database
+    include GameStart
+
+    START = 'start'
+    RULES = 'rules'
+    STATISTICS = 'stats'
+    EXIT = 'exit'
+
+    def welcome
+      puts I18n.t(:greeting)
+      run
+    end
+
+    def run
+      loop do
+        puts I18n.t(:menu)
+        case gets.chomp
+        when START then break registration
+        when RULES then rules
+        when STATISTICS then stats
+        when EXIT then break close
+        else puts I18n.t(:wrong_run)
+        end
+      end
+    end
+
+    def registration
+      GameConsole.new(choose_name, choose_difficulty).start
+      welcome
+    end
+
+    def choose_name
+      loop do
+        puts I18n.t(:choose_name)
+        name = gets.chomp
+        break close if name == 'exit'
+        break name if name_is_valid?(name)
+
+        puts I18n.t(:wrong_name)
+      end
+    end
+
+    def choose_difficulty
+      loop do
+        puts I18n.t(:choose_difficulty)
+        case gets.chomp
+        when 'exit' then break close
+        when 'easy' then break :easy
```

Comment:
can we use values from constant DIFFICULTY_LEVEL instead of strings?

Reply:
No. The value of this constant is [attempts 15, hints 3]

Reply:
`GameStart::DIFFICULTY_LEVEL.keys`

Reply:
i have rewritten code. i wrote:
def choose_difficulty
      loop do
        puts I18n.t(:choose_difficulty)
        input = gets.chomp
        break close if input == EXIT
        break input.to_sym if GameStart::DIFFICULTY_LEVEL.keys.include? input.to_sym
        puts I18n.t(:wrong_difficulty)
      end
    end

Reply:
Also you asked the value, not the key.